persistence-modules/hibernate-exceptions/src/test/java/com/baeldung/hibernate/exception/detachedentity/DetachedEntityUnitTest.java

```diff
@@ -0,0 +1,105 @@
+package com.baeldung.hibernate.exception.detachedentity;
+
+import com.baeldung.hibernate.exception.detachedentity.entity.Comment;
+import com.baeldung.hibernate.exception.detachedentity.entity.Post;
+
+import org.assertj.core.api.Assertions;
+import org.hibernate.Session;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.persistence.PersistenceException;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class DetachedEntityUnitTest {
+
+    private static Session session;
+    private Post detachedPost;
+
+    @Before
+    public void beforeEach() {
+        session = HibernateUtil.getSessionFactory()
+            .openSession();
+        session.beginTransaction();
+        this.detachedPost = new Post("Hibernate Tutorial");
+        session.persist(detachedPost);
+        session.evict(detachedPost);
+    }
+
+    @After
+    public void afterEach() {
+        clearDatabase();
+        session.close();
+    }
+
+    @Test
+    public void givenDetachedPost_whenTryingToPersist_thenThrowException() {
+        detachedPost.setTitle("Hibernate Tutorial for Absolute Beginners");
+
+        Assertions.assertThatThrownBy(() -> session.persist(detachedPost))
```

Comment:
Please use static imports for Assertions.

Reply:
I have added it now.